sled-agent/src/sim/instance.rs

```diff
@@ -305,25 +294,26 @@ impl SimInstanceInner {
 
     /// Returns the "resting" state the simulated instance will reach if its
     /// queue is drained.
-    fn next_resting_state(&self) -> ApiInstanceState {
+    fn next_resting_state(&self) -> VmmState {
         if self.queue.is_empty() {
             self.state.vmm().state
         } else {
             if let Some(last_state) = self.last_queued_instance_state() {
-                use ApiInstanceState as ApiState;
                 use PropolisInstanceState as PropolisState;
                 match last_state {
                     PropolisState::Creating | PropolisState::Starting => {
-                        ApiState::Starting
+                        VmmState::Starting
+                    }
+                    PropolisState::Running => VmmState::Running,
+                    PropolisState::Stopping => VmmState::Stopping,
+                    PropolisState::Stopped => VmmState::Stopped,
+                    PropolisState::Rebooting => VmmState::Rebooting,
+                    PropolisState::Migrating => VmmState::Migrating,
+                    PropolisState::Failed => VmmState::Failed,
+                    PropolisState::Destroyed => VmmState::Destroyed,
+                    PropolisState::Repairing => {
+                        unreachable!("Propolis doesn't use the Repairing state")
```

Comment:
this match is kind of making me wonder whether we really even need to have separate `PropolisInstanceState` and `VmmState` types --- they're _almost_ entirely the same except that `PropolisInstanceState` distinguishes between `Creating` and `Starting`, and has a `Repairing` state that apparently doesn't exist...

Reply:
I don't plan to make any changes along these lines in this PR, but a few things came to mind as I chewed on this comment:

- I think it's worthwhile to have some play in the joints between the way Omicron represents VMM states and the way Propolis represents its own state--they're currently designed to track each other very closely, but I think I'd rather have flexibility (of being able to change the types Propolis exposes without changing the Omicron data model) we don't need than to need flexibility we don't have.

- This specific case in the simulated sled agent is... weird. I'd say 90% of it is emulating Propolis's state machine, and 10% of it is enforcing rules about the state transitions Nexus is expected to request (see e.g. lines 106-118 of this file). I've long dreamt of getting rid of most of this logic and replacing it with a library/stub version of propolis-server's actual state machine (so that we don't have to keep the two state machines in sync) but haven't figured out how to do that yet.

- The Repairing state is indeed unused. We (Oxide writ large, not just the two of us) should revisit RFD 315 to figure out whether we want to have this state at all. If we don't need it, we should cut it from Propolis's API definition, after which it can disappear here. (I used `unreachable!` in this case because this is the simulation--IIRC in the real sled agent code we're prepared to accept a Repairing state from Propolis but will just convert it to a VMM state of Running.)